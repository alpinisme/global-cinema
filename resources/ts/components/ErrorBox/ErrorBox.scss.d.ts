// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'box': string;
  'message': string;
}
export const cssExports: CssExports;
export default cssExports;
