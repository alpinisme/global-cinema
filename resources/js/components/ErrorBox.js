import React from 'react';

const ErrorBox = ({ errors }) =>
    errors.map(e => (
        <ul key={e}>
            <li className="err-msg">{e}</li>
        </ul>
    ));

export default ErrorBox;
