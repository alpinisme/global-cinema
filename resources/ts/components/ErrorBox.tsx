import React, { ReactElement } from 'react';

const ErrorBox = (props: Props): ReactElement => {
    const list_items = props.errors.map(e => (
        <ul key={e}>
            <li className="err-msg">{e}</li>
        </ul>
    ));

    return <>{list_items}</>;
};

export default ErrorBox;

export interface Props {
    errors: string[];
}
