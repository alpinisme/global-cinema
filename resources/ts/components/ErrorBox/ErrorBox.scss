$err-color: rgb(156, 17, 40);

.message {
    color: $err-color;
}

.box {
    border: $err-color dashed 1px;
    padding: 0.5rem 1rem;
    width: 50%;
    margin: auto;
    list-style: none;
}
