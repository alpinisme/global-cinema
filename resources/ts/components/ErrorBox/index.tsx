import React, { ReactElement } from 'react';
import styles from './ErrorBox.scss';

const ErrorBox = ({ errors }: Props): ReactElement => {
    const errList = errors.map(e => (
        <ul key={e} className={styles.box}>
            <li className={styles.message} role="alert">
                {e}
            </li>
        </ul>
    ));

    return <>{errList}</>;
};

export default ErrorBox;

export interface Props {
    errors: string[];
}
