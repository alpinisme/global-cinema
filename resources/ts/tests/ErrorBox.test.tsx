import React from 'react';
import { create } from 'react-test-renderer';
import ErrorBox from '../components/ErrorBox';

describe('ErrorBox', () => {
    test('renders correctly', () => {
        const box = <ErrorBox errors={['error']} />;
        const tree = create(box).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('empty error array produces nothing', () => {
        const box = <ErrorBox errors={[]} />;
        const idea = create(box).root;
        expect(idea.children.length).toBe(0);
    });
});
