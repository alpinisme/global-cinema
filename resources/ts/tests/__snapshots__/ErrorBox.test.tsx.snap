// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox renders correctly 1`] = `
<ul
  className="box"
>
  <li
    className="message"
    role="alert"
  >
    error
  </li>
</ul>
`;
